migrator/migrations/m_187_to_m_188_remove_reports_without_migrated_collections/migration_impl.go

```diff
@@ -0,0 +1,89 @@
+package m187tom188
+
+import (
+	"context"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/stackrox/rox/migrator/migrations/m_187_to_m_188_remove_reports_without_migrated_collections/schema"
+	"github.com/stackrox/rox/pkg/sac"
+	"github.com/stackrox/rox/pkg/utils"
+	"gorm.io/gorm"
+)
+
+func migrate(db *gorm.DB) error {
+	ctx := sac.WithAllAccess(context.Background())
+
+	rows, err := db.WithContext(ctx).Table(schema.ReportConfigurationsTableName).Rows()
+	if err != nil {
+		return errors.Wrapf(err, "failed to iterate table %s", schema.ReportConfigurationsTableName)
+	}
+	if rows.Err() != nil {
+		utils.Should(rows.Err())
+		return errors.Wrapf(rows.Err(), "failed to get rows for %s", schema.ReportConfigurationsTableName)
+	}
+	defer func() { _ = rows.Close() }()
+
+	collectionsTable := db.WithContext(ctx).Table(schema.CollectionsTableName).Session(&gorm.Session{})
+
+	//var count int
+	var toDelete []schema.ReportConfigurations
+	for rows.Next() {
+		var config schema.ReportConfigurations
+		if err = db.ScanRows(rows, &config); err != nil {
+			return errors.Wrap(err, "failed to scan rows")
+		}
+		configProto, err := schema.ConvertReportConfigurationToProto(&config)
+		if err != nil {
+			return errors.Wrapf(err, "failed to convert %+v to proto", config)
+		}
+
+		//scope_id is not null then it is a V1 reportconfig
+		reportID := configProto.GetId()
+		scopeID := configProto.GetScopeId()
+		collectionID := configProto.GetResourceScope().GetCollectionId()
+
+		//scope_id is null/empty and collection_id is null/empty -> undefined, so delete
+		if strings.TrimSpace(scopeID) == "" && strings.TrimSpace(collectionID) == "" {
+			toDelete = append(toDelete, schema.ReportConfigurations{ID: reportID})
+			continue
+		}
+
+		// If scope_id is empty but collection_id is not, then it's a V2, so skip
+		// NOTE: This will skip even if the config has a scope_id (valid or invalid)
+		if strings.TrimSpace(collectionID) != "" {
```

Comment:
This is unexpected because migrations run against a specific db version, and for the version that introduces this migration, `collectionID`s are not expected. This will only happen if the transaction was corrupted (which shouldn't happen because that would mean we have a big problem within postgres lib, but let's consider postgres lib can have critical bugs).

You must check if the `collectionID` is present in the collection table.

Reply:
Are you suggesting that this migration should check for `collectionID` being valid (even though it's unlikely to exist and if it does, it is protected by a FK)? Or that regardless of if `collectionID` exists, we should validate that `scopeID` is pointing to a valid collection?

Reply:
> that regardless of if collectionID exists, we should validate that scopeID is pointing to a valid collection

Yes, because first `strings.TrimSpace(collectionID) != ""` is an unexpected condition for this migration and second, we still need to make sure `scopeID`, if non-empty, points to a valid collection because `scopeID` does not have fk constraint.